src/coreclr/nativeaot/Runtime/PalRedhawk.h

```diff
@@ -647,7 +647,9 @@ inline uint8_t * PalNtCurrentTeb()
 // conditional compilation (upto and including defining an export of functionality that isn't a supported
 // intrinsic on that platform).
 //
-
+#if defined(HOST_ARM)
+#define __cdecl
```

Comment:
Where do the other architectures (x64 or arm64) get the cdecl from?

Reply:
I assume this is MSVC specific calling convention, and probably this code was never compiled on non-Windows.

Reply:
I assume that mean I cannot find anything which make me think otherwise. I looking for gcc and clang but seems to be it's only MSVC specific

Reply:
Right, this file is included for Linux x64 or arm64 as well. Why is clang not complaining about this cdecl on existing Linux targets?

Reply:
This is what I still cannot get where this is included. 
I found following 3 files where __cdecl is redefined if needed
```
src\coreclr\pal\inc\pal_mstypes.h
src\coreclr\inc\palclr.h
src\coreclr\inc\coredistools.h
```
Last one seems to be not connected to building Runtime/Full, but other two can be somehow included. I try to track down 

I will continue hunt down, since on my Linux arm64 with command `./build.sh nativeaot -a arm64 -rc Debug -lc Release` I pass that point, so somewhere I should change macro conditions.

Side-node: Not sure if this is expected,
```
error MSB3030: Could not copy the file "~/runtimelab/artifacts/bin/coreclr/Linux.arm64.Debug/libjitinterface_arm64.so" because it was not found. [/~/runtimelab/src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj]```
```

Reply:
>  Could not copy the file "~/runtimelab/artifacts/bin/coreclr/Linux.arm64.Debug/libjitinterface_arm64.so"

It is not expected. We do official arm64 builds via cross-compilation on Linux x64 so it may explain why we do not see it.

Reply:
Things become more interesting. It seems that I know the reason why ARM not working.
Difference in compilation flags give me hint to the problem.
ARM
```
clang test.cpp -lstdc++ -fms-extensions -march=armv7-a
```
Literally anything else
```
clang test.cpp -lstdc++ -fms-extensions
```
Irregardless of `-march` presence it produce warning on ARM, and only way to suppress that is to undefine `__cdecl`.
So options are following:
- Should I place __cdecl augmentation in the `src/coreclr/nativeaot/Runtime/PalRedhawk.h` or `src/coreclr/nativeaot/Runtime/PalRedhawkCommon.h`
- Or maybe I can include `src\coreclr\pal\inc\pal_mstypes.h` or `src\coreclr\inc\palclr.h` in `PalRedhawkXXX.h`

Reply:
> Should I place __cdecl augmentation in the src/coreclr/nativeaot/Runtime/PalRedhawk.h or src/coreclr/nativeaot/Runtime/PalRedhawkCommon.h

Yes, that sounds good.